code/extensions/BasisFormProtectionExtension.php

```diff
@@ -2,25 +2,40 @@
 
 class BasisFormProtectionExtension extends Extension
 {
+	private static $controllers = array();
+	private static $includes = array();
 	private static $excludes = array();
 
 	public function updateForm()
 	{
-		if ($this->owner->controller instanceof LeftAndMain ||
-			$this->owner->controller instanceof TaskRunner ||
-			$this->owner->controller instanceof Security ||
-			$this->owner->controller instanceof DevelopmentAdmin ||
-			$this->owner->controller instanceof DevBuildController ||
-			$this->owner->controller instanceof DatabaseAdmin ||
-			$this->owner->controller instanceof GridFieldDetailForm_ItemRequest) {
-			return;
-		}
-		
-		if($this->owner->hasExtension('FormSpamProtectionExtension') && !$this->owner->isExcluded()) {
+		if($this->owner->hasExtension('FormSpamProtectionExtension') &&
+			($this->usesIncludesController() || $this->isIncluded()) &&
+			!$this->owner->isExcluded()) {
+			
 			$this->owner->enableSpamProtection();
 		}
 	}
 
+	public function usesIncludesController()
+	{
+		$controllers = Config::inst()->get(__CLASS__, 'controllers');
+		foreach(ClassInfo::ancestry($this->owner->controller) as $controller) {
```

Comment:
`controller` is a protected variable on a Form object - if this is working, it's probably only because it's abusing SS's __get mechanism. Can this please use `getController()` ?